src/pages/stats/[login].tsx

```diff
@@ -28,30 +28,38 @@ export default function Stats() {
   const [searchQuery, setSearchQuery] = useState<string>("");
 
   const filteredRepositories = useMemo(() => {
-    if (!searchQuery) {
-      return hideOwnRepo
-        ? repositories?.filter(
-            // apply hideOwnRepo filter if checked
-            (repoData) =>
-              repoData.repository.owner.login !== session?.user.login
-          )
-        : repositories;
-    }
+    const filterOutOwnRepos = (
+      repositories: PullRequestContributionsByRepository[]
+    ) => {
+      return repositories.filter(
+        (repoData) => repoData.repository.owner.login !== session?.user.login
+      );
+    };
+
+    const filterReposBySearchQuery = (
+      repositories: PullRequestContributionsByRepository[]
+    ) => {
+      const query = searchQuery.toLowerCase();
+      return repositories?.filter((repoData) =>
+        repoData.repository.name.toLowerCase().includes(query)
+      );
+    };
+
+    const filterRepos = (
+      repositories: PullRequestContributionsByRepository[]
+    ) => {
+      if (!searchQuery) {
+        return hideOwnRepo ? filterOutOwnRepos(repositories) : repositories;
+      }
+
+      const filteredReposBySearchQuery = filterReposBySearchQuery(repositories);
+      const filteredOutOwnRepos = filterOutOwnRepos(filteredReposBySearchQuery);
 
-    // filter repositories based on search query
-    const query = searchQuery.toLowerCase();
-    const filterRepos = repositories?.filter((repoData) =>
-      repoData.repository.name.toLowerCase().includes(query)
-    );
+      return hideOwnRepo ? filteredOutOwnRepos : filteredReposBySearchQuery;
```

Comment:
Here you're running `filterOutOwnRepos` in any case, but if `hideOwnRepo` is false it is unnecessary.

I'd probably see this better with an if statement to return early, or if we want to keep it more functional with the ternary you should run `filterOutOwnRepos` directly inside the "then" of the ternary.